NoSuchFileException when ~/.njord is missing

When running njord for the first time, it complains about lack of ~/.njord directory. This directory should be created automatically if it doesn't exist.

`mvnd deploy`
```
[INFO] --- deploy:3.1.4:deploy (default-deploy) @ deploytest ---
[INFO] --------------------------------------------------------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] --------------------------------------------------------------------------------------------------------------------------
[INFO] Total time:  0.757 s (Wall Clock)
[INFO] Finished at: 2025-05-11T19:19:08-07:00
[INFO] --------------------------------------------------------------------------------------------------------------------------
[INFO] Njord session closed
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-deploy-plugin:3.1.4:deploy (default-deploy) on project deploytest: Execution default-deploy of goal org.apache.maven.plugins:maven-deploy-plugin:3.1.4:deploy failed: java.nio.file.NoSuchFileException: /Users/lprimak/.njord -> [Help 1]
```

--- a/core/src/main/java/eu/maveniverse/maven/njord/shared/SessionConfig.java
+++ b/core/src/main/java/eu/maveniverse/maven/njord/shared/SessionConfig.java
@@ -426,7 +426,15 @@ public interface SessionConfig {
                 this.enabled = enabled;
                 this.dryRun = dryRun;
                 this.version = version;
-                this.basedir = requireNonNull(basedir, "basedir");
+                this.basedir = getCanonicalPath(basedir);
+                if (!Files.isDirectory(this.basedir)) {
+                    try {
+                        Files.createDirectories(this.basedir);
+                    } catch (IOException e) {
+                        throw new UncheckedIOException("Cannot create basedir", e);
+                    }
+                }
+
                 requireNonNull(propertiesPath, "propertiesPath");
                 this.propertiesPath = getCanonicalPath(basedir.resolve(propertiesPath));
 
@@ -435,7 +443,7 @@ public interface SessionConfig {
                     try (InputStream inputStream = Files.newInputStream(this.propertiesPath)) {
                         properties.load(inputStream);
                     } catch (IOException e) {
-                        throw new UncheckedIOException(e);
+                        throw new UncheckedIOException("Cannot load properties", e);
                     }
                 }
 

